wgpu-core/src/hub.rs

```diff
@@ -16,21 +16,42 @@ use wgt::Backend;
 use std::cell::Cell;
 use std::{fmt::Debug, marker::PhantomData, mem, ops};
 
-/// A simple structure to manage identities of objects.
+/// A simple structure to allocate [`Id`] identifiers.
+///
+/// Calling [`alloc`] returns a fresh, never-before-seen id. Calling [`free`]
+/// marks an id as dead; it will never be returned again by `alloc`.
+///
+/// Use `IdentityManager::default` to construct new instances.
+///
+/// The index values of `Id`s allocated by an `IdentityManager` are suitable for
+/// use as indices into a vector of information about the ids' referents. The
+/// `alloc` method returns an id with a given index only if there is some extant
```

Comment:
I don't understand this sentence. The method returns a given index only if there is some higher index in use?
Also, "ll lower non-negative index values are currently in use" isn't entirely correct. There is a freelist inside, and it gives you the last index that was freed, which isn't necessarily the lowest free index.

Reply:
Right, I was trying to capture the behavior of the freelist there. It's too dense, I'll rephrase.